### PR TITLE
[type] [bug] Fix global load of CustomFloatType on CUDA

### PR DESCRIPTION
**Pull request by Hanke98**
Related issue = #1905 #2065

Some mistakes were made in global loading on CUDA, which would lead to compilation error. I will point out the bug below.

And also, I added a new test case focusing on this problem in this pr 

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

File: taichi/backends/cuda/codegen_cuda.cpp [hunk 1]
```diff
@@ -425,19 +425,20 @@ class CodeGenLLVMCUDA : public CodeGenLLVM {
...
+          Type* int_in_mem = nullptr;
           if (auto cit = val_type->cast<CustomIntType>()) {
+            int_in_mem = val_type;
             dtype = cit->get_physical_type();
           } else if (auto cft = val_type->cast<CustomFloatType>()) {
-            dtype = cft->get_compute_type()
-                        ->as<CustomIntType>()
-                        ->get_physical_type();
+            int_in_mem = cft->get_digits_type();
+            dtype = int_in_mem->as<CustomIntType>()->get_physical_type();
```
Comment (Hanke98) [inline taichi/backends/cuda/codegen_cuda.cpp:434]: The `compute_type` of `CustomFloatType` is f32, f64,... which cannot be `CustomIntType`

File: tests/python/test_custom_float.py [hunk 2]
```diff
@@ -51,3 +51,22 @@ def rotate_18_degrees():
...
+    ci15 = ti.type_factory_.get_custom_int_type(15, True)
+    cft = ti.type_factory_.get_custom_float_type(ci15, ti.f32.get_ptr(), 0.1)
+    x = ti.field(dtype=cft)
+
+    ti.root._bit_struct(num_bits=32).place(x)
+
+    @ti.kernel
+    def test(data: ti.f32):
+        ti.cache_read_only(x)
+        assert x[None] == data
```
Comment (Hanke98) [inline tests/python/test_custom_float.py:67]: Force the `x` cache read only, to test global loading in `codegen_cuda.cpp`. Or, the `x` might be read the function in `codegen_llvm.cpp`

File: taichi/backends/cuda/codegen_cuda.cpp [hunk 3]
```diff
@@ -425,19 +425,20 @@ class CodeGenLLVMCUDA : public CodeGenLLVM {
           auto val_type = ptr_type->get_pointee_type();
           llvm::Value *data_ptr = nullptr;
           llvm::Value *bit_offset = nullptr;
+          Type *int_in_mem = nullptr;
```
Comment (yuanming-hu) [inline taichi/backends/cuda/codegen_cuda.cpp:428]:
```suggestion
          Type *int_in_mem = nullptr;
          // For CustomIntType "int_in_mem" refers to the type itself;
          // for CustomFloatType "int_in_mem" refers to the CustomIntType of the digits. 
```

**Review by yuanming-hu — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**